From f35a8efe2c34496eaf45f8a9bd5bb7625d34bb5c Mon Sep 17 00:00:00 2001
From: Mario Limonciello <mario.limonciello@dell.com>
Date: Wed, 1 Nov 2017 14:25:22 -0500
Subject: [PATCH] platform/x86: dell-smbios: Prefix class/select with cmd_
Git-commit: f35a8efe2c34496eaf45f8a9bd5bb7625d34bb5c
Patch-mainline: v4.15-rc1
References: FATE#325842

Later on these structures will be brought up to userspace.
the word "class" is a reserved word in c++ and this will prevent
uapi headers from being included directly in c++ programs.

To make life easier on these applications, prepare the change now.

Signed-off-by: Mario Limonciello <mario.limonciello@dell.com>
Reviewed-by: Edward O'Callaghan <quasisec@google.com>
Signed-off-by: Darren Hart (VMware) <dvhart@infradead.org>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/platform/x86/dell-smbios.c |    4 ++--
 drivers/platform/x86/dell-smbios.h |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/platform/x86/dell-smbios.c
+++ b/drivers/platform/x86/dell-smbios.c
@@ -86,8 +86,8 @@ void dell_smbios_send_request(int class,
 	command.ebx = virt_to_phys(buffer);
 	command.ecx = 0x42534931;
 
-	buffer->class = class;
-	buffer->select = select;
+	buffer->cmd_class = class;
+	buffer->cmd_select = select;
 
 	dcdbas_smi_request(&command);
 }
--- a/drivers/platform/x86/dell-smbios.h
+++ b/drivers/platform/x86/dell-smbios.h
@@ -22,8 +22,8 @@ struct notifier_block;
  * system management mode, hence the volatiles */
 
 struct calling_interface_buffer {
-	u16 class;
-	u16 select;
+	u16 cmd_class;
+	u16 cmd_select;
 	volatile u32 input[4];
 	volatile u32 output[4];
 } __packed;
